kernel/stack.c

```diff
@@ -85,9 +86,9 @@ Z_SYSCALL_HANDLER(k_stack_alloc_init, stack, num_entries)
 
 void k_stack_cleanup(struct k_stack *stack)
 {
-	__ASSERT_NO_MSG(!_waitq_head(&stack->wait_q));
+	__ASSERT_NO_MSG(_waitq_head(&stack->wait_q) == NULL);
 
-	if (stack->flags & K_STACK_FLAG_ALLOC) {
+	if ((stack->flags & K_STACK_FLAG_ALLOC) != (u8_t)0U) {
```

Comment:
Why the typecast and the u suffix?  There's absolutely no reason not to write "!= 0" here.  The left hand side will promote to "int" from "unsigned char" perfectly without worries about sign or width issues.

Reply:
Actually only the typecast is needed. "U" is unnecessary. updated the patch.